src/subdomains/supporting/pricing/utils/price-step.ts

```diff
@@ -9,30 +9,28 @@ import { PricingUtil } from './pricing.util';
 export interface PriceStepOptions {
   from?: string | 'input';
   to?: string | 'output';
-  overwriteReferenceTo?: string;
-  fallbackPrimaryTo?: string;
-  providers?: PriceStepProviders;
+  primary?: PriceStepProviderOptions;
+  reference?: PriceStepProviderOptions;
   fixedPrice?: number;
 }
 
-export interface PriceStepProviders {
-  primary: PriceProvider[];
-  reference: PriceProvider[];
+export interface PriceStepProviderOptions {
+  providers?: PriceProvider[];
```

Comment:
I think providers must be required. Once someone configures primary or reference, he needs to specify providers. For example for primary providers if nothing configured, init specification will not pass it through at server start, though compile-wise everything will look good

Reply:
however you default to empty lists... yeah, init will actually pass, up to you

Reply:
Ok agreed. `primary` or `reference` are not required, but `providers` should be 👍